packages/engine-core/src/binary/Connection.ts

```diff
@@ -1,14 +1,17 @@
 import getStream from 'get-stream'
-import type { Client } from 'undici'
-import { Pool } from 'undici'
+import type { Dispatcher, Pool } from 'undici'
 import type { URL } from 'url'
 
 export type Result<R> = {
-  statusCode: Client.ResponseData['statusCode']
-  headers: Client.ResponseData['headers']
+  statusCode: Dispatcher.ResponseData['statusCode']
+  headers: Dispatcher.ResponseData['headers']
   data: R
 }
 
+// because undici lazily loads llhttp wasm which bloats the memory
+// TODO: hopefully replace with `import` but that causes segfaults
+const undici = () => require('undici')
```

Comment:
So that's our current workaround for the segfaults for the Binary? any idea why is this happening?

Reply:
I initially wanted to do `() => import('undici')` and I could not understand why this could cause segfaults in `jest` and only on CI. So I swapped for `() => require('undici')`.

Reply:
Worth opening a PR again to see if it still happens, and if so maybe point it out to Undici/Node ppl?

Reply:
I tried this today. I made `undici` not to load lazily like it does by re-wrapping it in `() => `. Somehow, their lazy loading of the wasm (ie. on module load) was also causing memory leaks in jest. It would make sense to debug where it segfaults when doing `import` instead of `require` so that we can report to them.